<!DOCTYPE html>
<html lang="en" class="html" data-theme="dark"><head>
  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
  
  <script> 
  MathJax = {
    tex: {
      inlineMath: [['$', '$']],
      processEscapes: true
    }
  };
  </script>

  <title>
    
      Lecture 10_ regulatory motifs
    
  </title>

  <!-- Begin Jekyll SEO tag v2.7.3 -->
<meta name="generator" content="Jekyll v4.2.2" />
<meta property="og:title" content="Lecture 10_ regulatory motifs" />
<meta name="author" content="zying" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Lecture 10: Regulator Motifs Discovery Features" />
<meta property="og:description" content="Lecture 10: Regulator Motifs Discovery Features" />
<link rel="canonical" href="http://localhost:4000/lecture-10-regulatory-motifs" />
<meta property="og:url" content="http://localhost:4000/lecture-10-regulatory-motifs" />
<meta property="og:site_name" content="Boundless" />
<meta property="og:image" content="https://soopr.xyz/images/card?url=http://localhost:4000/lecture-10-regulatory-motifs" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2023-02-18T00:00:00+08:00" />
<meta name="twitter:card" content="summary_large_image" />
<meta property="og:image" content="https://soopr.xyz/images/card?url=http://localhost:4000/lecture-10-regulatory-motifs" />
<meta property="twitter:title" content="Lecture 10_ regulatory motifs" />
<script type="application/ld+json">
{"url":"http://localhost:4000/lecture-10-regulatory-motifs","headline":"Lecture 10_ regulatory motifs","dateModified":"2023-02-18T00:00:00+08:00","datePublished":"2023-02-18T00:00:00+08:00","@type":"BlogPosting","author":{"@type":"Person","name":"zying"},"description":"Lecture 10: Regulator Motifs Discovery Features","mainEntityOfPage":{"@type":"WebPage","@id":"http://localhost:4000/lecture-10-regulatory-motifs"},"@context":"https://schema.org"}</script>
<script async defer data-soopr-token="" src="https://sdk.soopr.co/soopr.js"  ></script>
<!-- End Jekyll SEO tag -->

  <link type="application/atom+xml" rel="alternate" href="http://localhost:4000/feed.xml" title="Boundless" />

  <!-- Favicon -->
  <link rel="apple-touch-icon" sizes="180x180" href="/assets/images/favicon/apple-touch-icon.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/assets/images/favicon/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/assets/images/favicon/favicon-16x16.png">
  <link rel="manifest" href="/assets/images/favicon/site.webmanifest">
  <link rel="mask-icon" href="/assets/images/favicon/safari-pinned-tab.svg" color="#5bbad5">
  <link rel="shortcut icon" href="/assets/images/favicon/favicon.ico">
  <meta name="msapplication-TileColor" content="#00aba9">
  <meta name="msapplication-config" content="/assets/images/favicon/browserconfig.xml">
  <meta name="theme-color" content="#ffffff">
  <!-- Favicon -->

  <link rel="stylesheet" href="/assets/css/main.css" />
  
    <script type="text/javascript">
  window.addEventListener('load', themeChange);
  const currentTheme = localStorage.getItem('theme') ? localStorage.getItem('theme') : null;
  if (currentTheme)
    document.documentElement.setAttribute('data-theme', currentTheme);

  function themeChange() {
    let button = document.querySelector('.theme-toggle');

    button.addEventListener('click', function (e) {
      let currentTheme = document.documentElement.getAttribute('data-theme');
      if (currentTheme === 'dark') {
        transition();
        document.documentElement.setAttribute('data-theme', 'light');
        localStorage.setItem('theme', 'light');
      } else {
        transition();
        document.documentElement.setAttribute('data-theme', 'dark');
        localStorage.setItem('theme', 'dark');
      }
    });

    let transition = () => {
      document.documentElement.classList.add('transition');
      window.setTimeout(() => {
        document.documentElement.classList.remove('transition');
      }, 1000);
    }
  }
</script>


  
</head>
<body>
    <main class="page-content" aria-label="Content">
      <div class="w">
        <a href="/">home..</a>
<h1 class="post-title">Lecture 10_ regulatory motifs</h1>
<p class="post-date text-bold">
  
    <span class="text-upcase">zying</span> / 
  
  
    <span class="text-upcase">February 2023</span>
  


  
  
  (385 Words, 
  3 Minutes)
  

</p>

<div class="soopr-btn"
   data-twitter="SooprCo"
>
</div>


  <div class="">
    
    <span class="tag">Lectures</span>
    
    <span class="tag">ML_in_genomics</span>
    
  </div>


<h1 id="lecture-10-regulator-motifs-discovery">Lecture 10: Regulator Motifs Discovery</h1>
<h2 id="features">Features</h2>

<ul>
  <li>turned on/off response to changing environment</li>
  <li>no direct addressing: genes contain sequences of motifs (tags)</li>
  <li>specialized protein (TF) recognize these tags</li>
  <li>motif has many different class (enhancer, promoter, splicing…)</li>
  <li>not limited to DNA sequences (spicing signals, Domains and epitopes at protein level, recurring patterns at physiological level…)
    <h3 id="regulator-structure-to-recognize-motifs">Regulator structure to recognize motifs</h3>
  </li>
  <li>proteins chemical properties can recognized DNA fregments (don’t need open DNA)</li>
  <li>3D topology dictates specificity (fully constrained pos or ambiguous/degenerate pos)</li>
  <li>other types of recognition (micro RNA, Nucleosomes, RNAs structure)
    <h3 id="infomatics-perspect-of-motif">Infomatics perspect of motif</h3>
  </li>
  <li>summary information</li>
  <li>integrate many pos</li>
  <li>measure of information</li>
</ul>

<p>(generative model)</p>
<h2 id="challenges-of-motifs-discovery">Challenges of motifs discovery</h2>

<ul>
  <li>short (6-8 bp), sometimes degenerate</li>
  <li>can contain any set of nucleotides</li>
  <li>Act at variable distance (upstream of downstream)</li>
</ul>

<p><img src="images/image_8.png" alt="image.png" /></p>
<h2 id="general-thoughts-of-motifs-discovery">General thoughts of motifs discovery</h2>

<ul>
  <li>Two assumptions about the data
    <ul>
      <li>assume no pairwise correlations between bases (each base is independent) [the risk of overfitting data]</li>
      <li>all motifs have fixed lengths</li>
    </ul>
  </li>
  <li>methods for motif discovery
    <ul>
      <li>region-based (EM, Gibbs sampling, greedy algorithm…)
        <ul>
          <li>local alignment</li>
          <li>find non-random seqences</li>
          <li>reduce the search space by applying prior knowledge</li>
          <li>conserved blocks</li>
          <li>examine the requency of kmers</li>
          <li>probabilistic methods</li>
        </ul>
      </li>
      <li>genome-wide (conservation-based)</li>
      <li>In vitro/trans (protein domains): based experiments</li>
    </ul>
  </li>
</ul>

<p><img src="images/image_9.png" alt="image.png" /></p>
<h1 id="expectation-maximization">Expectation maximization</h1>
<h2 id="key-idea-in-em">Key idea in EM</h2>

<ul>
  <li>Starting positions&lt;-&gt;motif matrix
    <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Basic Iterative Approach:
Given: length parameter W, traning <span class="nb">set </span>of sequences
    	<span class="nb">set </span>initial value <span class="k">for </span>motif
    	<span class="k">do</span> 
      	-&gt; re-estimate starting-position from motif
      	-&gt; re-estimate motif from starting-position
    	<span class="k">until </span>convergence
    	<span class="k">return</span>: motif,starting positions
</code></pre></div>    </div>
  </li>
  <li>EM is a deterministic algorithm: dependent on the initial starting points, maybe converged in local max but not global</li>
  <li>rerun the algorithm with different initial starting positions to try reduce the chance of converging on local max (MEME-Multiple EM for motif Elicitation)
    <h2 id="the-e-step-estimating-z_ij-from-the-pwm-profile-matrix">The E step: estimating $Z_ij$ from the PWM (profile matrix)</h2>
  </li>
  <li><strong>Initialization</strong> (generate an initial probability weight matrix)
    <ul>
      <li>can initialize the PWM by choosing starting locations randomly</li>
      <li>if there is 0 probability, it is generally good idea to insert pseudo- counts into your probabilities</li>
      <li>keep a background distribution</li>
    </ul>
  </li>
  <li><strong>Expectation</strong> (generate a vector $Z_ij$)
    <ul>
      <li>calculate $Z_ij$using Bayes’ Rule (can’t comprehense clearly)</li>
    </ul>
  </li>
</ul>

<p><strong>M step:</strong> Finding the maximum likelihood motif from starting positions $Z_ij$</p>

<ul>
  <li><strong>Maximization</strong></li>
  <li><strong>Repeat</strong>
    <ul>
      <li>to measure how much each element in the PWM changes after step max is a possible way to test whether the profile matrix has converged
        <h1 id="gibbs-sampling-sample-from-joint-mz_ij">Gibbs Sampling: Sample from joint $(M,Z_ij)$</h1>
      </li>
    </ul>
  </li>
  <li>Sampling motif position based on the Z vector
    <ul>
      <li><strong>Initialization</strong></li>
      <li><strong>Remove</strong></li>
      <li><strong>Update</strong></li>
      <li><strong>Sample</strong></li>
      <li><strong>Iterate</strong></li>
    </ul>
  </li>
  <li>More likely to find global maximum, easy to implement</li>
</ul>

<h1 id="de-novo-motif-discovery">De novo motif discovery</h1>
<h2 id="motivation-for-de-novo-motif-discovery">Motivation for de novo motif discovery</h2>

<ul>
  <li>TF and centric approaches are not comprehensive and are biased</li>
  <li>generally require TF (antibody to factor)</li>
  <li>De novo using conservation is unbiased, <font color="red"> but can't match motif to factor and require multiple genomes</font>
    <h2 id="using-genome-wide-conservation">Using genome-wide conservation</h2>
  </li>
</ul>

<h2 id="validation-of-discovered-motifs-with-functional-datasets">Validation of discovered motifs with functional datasets</h2>

<h1 id="evolutionary-signatures-for-regulatory-motifs">Evolutionary signatures for regulatory motifs</h1>

<h1 id="de-novo-dissection-and-confirmation-of-regulaory-regions">De novo Dissection and confirmation of regulaory regions</h1>
<p>#</p>
<h1 id="others">Others</h1>
<h2 id="possibly-deprecated-stuff">Possibly deprecated stuff</h2>

<h2 id="comparing-different-methods">Comparing different Methods</h2>

<h2 id="oops-zoops-tcm">OOPS, ZOOPS, TCM</h2>

<h2 id="motif-representation-and-information-content">Motif Representation and Information content</h2>


        
          <button title="Toggle Theme" class="theme-toggle">
  <svg viewBox="0 0 32 32" width="24" height="24" fill="currentcolor">
    <circle cx="16" cy="16" r="14" fill="none" stroke="currentcolor" stroke-width="4"></circle>
    <path d="
             M 16 0
             A 16 16 0 0 0 16 32
             z">
    </path>
  </svg>
</button>

        
        <div class="credits">&copy;&nbsp;2023&nbsp;zying
          &nbsp;
          •
          &nbsp;Powered by <a href="https://www.soopr.co" target="_blank" rel="noreferrer">Soopr</a>
          &nbsp;
          •
          &nbsp;Theme&nbsp; <a href="https://github.com/abhinavs/moonwalk" target="_blank" rel="noreferrer">Moonwalk</a>
        </div>
      </div>
    </main><script async defer src="https://sdk.soopr.co/soopr.js"></script></body>
</html>
